### PR TITLE
Add ec2 workflow for ci

### PR DESCRIPTION
**Pull request by potsrevennil**

### CODE REVIEW

File: .github/workflows/bench.yml [hunk 1]
```diff
@@ -45,3 +46,38 @@ jobs:
...
+            cflags: -DFORCE_AARCH64
+    uses: ./.github/workflows/bench_ec2_reusable.yml
+    if: github.repository_owner == 'pq-code-package' && (github.event.label.name == 'benchmark' || github.ref == 'refs/heads/main')
+    with:
+      ec2_instance_type: ${{ matrix.target.ec2_instance_type }}
+      ec2_ami: ${{ matrix.target.ec2_ami }}
+      archflags: ${{ matrix.target.archflags }}
+      cflags: ${{ matrix.target.cflags }}
+      opt: ${{ matrix.opt.value }}
+      store_results: ${{ matrix.opt.value }} # Only store optimized results
```
Comment (mkannwischer) [inline .github/workflows/bench.yml:81]: store results should only be true if it's the main branch, right?

**Review by mkannwischer — DISMISSED** [review]

File: .github/workflows/bench.yml (hunk 1 above)
Comment (potsrevennil) [reply .github/workflows/bench.yml:81]: This is related to https://github.com/pq-code-package/mlkem-c-aarch64/pull/126/files

It was due to that we didn't disambiguate opt vs. non-opt in the stats pushed to github pages.

Comment (mkannwischer) [reply .github/workflows/bench.yml:81]: that's okay, but it should still only store it on the main branch
currently it creates a lot of commit for G2 and G3 on the PRs - that is inconsistent with the other benchmarks.

Comment (potsrevennil) [reply .github/workflows/bench.yml:81]: oh yes! your're right

**Comment by mkannwischer** [comment]
Sorry you will have to rebase this after #139  got merged.

File: .github/workflows/bench.yml (hunk 1 above)
Comment (hanno-becker) [reply .github/workflows/bench.yml]: @potsrevennil Can you open a separate PR for this?

Comment (potsrevennil) [reply .github/workflows/bench.yml]: ok

**Review by hanno-becker — APPROVED** [review]

**Pull request merged**